src/Tribe/Utils/Post_Thumbnail.php

```diff
@@ -155,44 +157,58 @@ public function fetch_data() {
 			return $this->data;
 		}
 
-		$post_id      = $this->post_id;
 		$image_sizes  = $this->get_image_sizes();
 		$thumbnail_id = $this->thumbnail_id;
 
-		$thumbnail_data = array_combine(
-			$image_sizes,
-			array_map(
-				static function ( $size ) use ( $thumbnail_id ) {
-					$size_data = wp_get_attachment_image_src( $thumbnail_id, $size );
+		$cache_key = empty( $thumbnail_id ) ? -1 : $thumbnail_id;
+
+		if ( empty( $cache_thumbnail[ $cache_key ] ) ) {
+			$thumbnail_data = array_combine(
+				$image_sizes,
+				array_map(
+					static function( $size ) use ( $thumbnail_id ) {
+						static $cache_size_data = [];
+
+						$size_data_cache_key = empty( $thumbnail_id ) ? -1 : $thumbnail_id;
+						$size_data_cache_key = "{$size_data_cache_key}:{$size}";
+
+						if ( ! isset( $cache_size_data[ $size_data_cache_key ] ) ) {
+							$cache_size_data[ $size_data_cache_key ] = wp_get_attachment_image_src( $thumbnail_id, $size );
+						}
+
+						$size_data = $cache_size_data[ $size_data_cache_key ];
+
+						if ( false === $size_data ) {
+							return (object) [
+								'url'             => '',
+								'width'           => '',
+								'height'          => '',
+								'is_intermediate' => false,
+							];
+						}
 
-					if ( false === $size_data ) {
 						return (object) [
-							'url'             => '',
-							'width'           => '',
-							'height'          => '',
-							'is_intermediate' => false,
+							'url'             => Arr::get( $size_data, 0, '' ),
+							'width'           => Arr::get( $size_data, 1, '' ),
+							'heigth'          => Arr::get( $size_data, 2, '' ),
+							'is_intermediate' => (bool) Arr::get( $size_data, 3, false ),
 						];
-					}
+					},
+					$image_sizes
+				)
+			);
 
-					return (object) [
-						'url'             => Arr::get( $size_data, 0, '' ),
-						'width'           => Arr::get( $size_data, 1, '' ),
-						'heigth'          => Arr::get( $size_data, 2, '' ),
-						'is_intermediate' => (bool) Arr::get( $size_data, 3, false ),
-					];
-				},
-				$image_sizes
-			)
-		);
+			$srcset                   = wp_get_attachment_image_srcset( $thumbnail_id );
+			$thumbnail_data['srcset'] = ! empty( $srcset ) ? $srcset : false;
 
-		$srcset                   = wp_get_attachment_image_srcset( $thumbnail_id );
-		$thumbnail_data['srcset'] = ! empty( $srcset ) ? $srcset : false;
+			$title                   = get_the_title( $thumbnail_id );
+			$thumbnail_data['title'] = ! empty( $title ) ? $title : false;
 
-		$title                   = get_the_title( $thumbnail_id );
-		$thumbnail_data['title'] = ! empty( $title ) ? $title : false;
+			$alt                   = trim( strip_tags( get_post_meta( $thumbnail_id, '_wp_attachment_image_alt', true ) ) );
```

Comment:
:warning: `strip_tags()` does not strip CSS and JS in between the script and style tags. Use `wp_strip_all_tags()` to strip all tags.

<sub>hash: f10bafcb6e1cdc43cbf646e1742dbcdf</sub>

Reply:
Should this use `sanitize_text_field()` instead?

Reply:
```suggestion
			$alt                   = sanitize_text_field( get_post_meta( $thumbnail_id, '_wp_attachment_image_alt', true ) );
```

Reply:
Perhaps, but due to the current timeline, I'll avoid making changes unrelated to performance for now. I'll ticket this in Jira, though! Good catch.

Reply:
https://moderntribe.atlassian.net/browse/BTRIA-110